github/pull_request.rb

```diff
@@ -187,9 +187,17 @@ def add_status(state, message, options = {})
     @client.create_status(repo.full_name, options[:sha], state, context: 'prprocessor', description: message, target_url: options[:url])
   end
 
-  def set_directory_labels(mapping)
+  def get_labels(filename, mapping)
+    mapping.select { |k, v| filename == k || filename.start_with?((k+"/").gsub(/\/+$/, "/")) }.values
```

Comment:
I don't like the double trailing slash cleaning here. Can't we just require the configuration to be correct? We can introduce a test on `config/repos.yaml` that fails if there's a trailing slash in a mapping. That way Travis will fail on a PR that attempts to introduce an invalid config.